src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
+   * @inner
+   */
+  getAuthToken(): string | null {
+    return this.authToken;
+  }
+
   /**
    * When the wallet starts, it might take some seconds for the wallet service to completely load all addresses
    * This method is responsible for polling the wallet status until it's ready
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
   async startPollingStatus() {
     try {
-      const res = await walletApi.getWalletStatus(this.walletId!);
+      const res = await walletApi.getWalletStatus(this);
       const data = res.data;
       if (res.status === 200 && data.success) {
         if (data.status.status === 'creating') {
           setTimeout(async () => {
             await this.startPollingStatus();
           }, WALLET_STATUS_POLLING_TIMEOUT);
         } else if (data.status.status === 'ready') {
-          this.setState(walletState.READY);
+          await this.onWalletReady();
         } else {
-          // TODO What other status might have?
-          // throw error?
+          throw new WalletRequestError('Error getting wallet status.');
         }
       } else {
-        // TODO is there any other possible error to handle here?
+        throw new WalletRequestError('Error getting wallet status.');
       }
     } catch(err) {
-      // TODO How to handle error
-      console.log('Error sending create wallet request', err);
+      // Should we handle any error?
+      throw err;
+    }
+  }
+
+  /**
+   * Check if wallet is ready and throw error if not ready
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  private checkWalletReady() {
+    if (!this.isReady()) {
+      throw new WalletError('Wallet not ready');
     }
   }
 
+  /**
+   * Method executed when wallet is ready
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  private async onWalletReady() {
+    this.setState(walletState.READY);
+    await this.getNewAddresses();
+  }
+
   /**
    * Get all addresses of the wallet
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getAllAddresses(): Promise<string[]> {
-    const response = await walletApi.getAddresses(this.walletId!);
-    let addresses = [];
+  async getAllAddresses(): Promise<GetAddressesObject[]> {
+    this.checkWalletReady();
+    const response = await walletApi.getAddresses(this);
+    let addresses: GetAddressesObject[] = [];
     if (response.status === 200 && response.data.success === true) {
       addresses = response.data.addresses;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet addresses.');
     }
     return addresses;
   }
 
+  private async getNewAddresses() {
+    this.checkWalletReady();
+    const response = await walletApi.getNewAddresses(this);
+    let addresses: AddressInfoObject[] = [];
+    if (response.status === 200 && response.data.success === true) {
```

Comment:
Shouldn't we move this logic to the walletApi? I think this code must be way simpler to read because it would have to deal with a promise that might succeed or fail.

Reply:
Makes sense, done.

Reply:
This refactor is great. It is way easier to read the code. Thanks!